File: Quokka.RISCV.CS2CPP.Tests/Intermediate/FirmwareSource/DataDeclarationTest.cpp
#include "DataDeclarationTest.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace DataDeclarationTestSource
{
	void Firmware::EntryPoint()
	{
		unsigned char b1 = 10, b2 = 20;
		int i1 = 0, i2 = 100;
		unsigned int ui1 = 10000;
		DataDeclarationTestSource_SOC_Result = (int)(((((b1 + b2)) * ((i1 - i2))) + ui1));
	}
}

File: Quokka.RISCV.CS2CPP.Tests/Source/DataDeclarationTest.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace DataDeclarationTestSource
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual int Result { get; set; }
    }

    public static class Firmware
    {
        public static void EntryPoint()
        {
            byte b1 = 10, b2 = 20;
            int i1 = 0, i2 = 100;
            uint ui1 = 10000;

            SOC.Instance.Result = (int)((b1 + b2) * (i1 - i2) + ui1);
        }
    }
}
